From 906d0c1381e865dc7c96a4bde6fe694f1ce089a9 Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Thu, 30 Jul 2015 16:35:52 -0500
Subject: [PATCH] MLK-11335: mtd: qspi: enalbe LPSR for qspi

The LPSR turns off the power for IOMUX when suspending so restore the
IOMUX when resuming.

Signed-off-by: Han Xu <b45815@freescale.com>
---
 drivers/mtd/spi-nor/fsl-quadspi.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mtd/spi-nor/fsl-quadspi.c b/drivers/mtd/spi-nor/fsl-quadspi.c
index 7ab261f..3facf0f 100644
--- a/drivers/mtd/spi-nor/fsl-quadspi.c
+++ b/drivers/mtd/spi-nor/fsl-quadspi.c
@@ -1225,6 +1225,7 @@ static int fsl_qspi_remove(struct platform_device *pdev)
 
 static int fsl_qspi_suspend(struct platform_device *pdev, pm_message_t state)
 {
+	pinctrl_pm_select_sleep_state(&pdev->dev);
 	return 0;
 }
 
@@ -1233,6 +1234,8 @@ static int fsl_qspi_resume(struct platform_device *pdev)
 	int ret;
 	struct fsl_qspi *q = platform_get_drvdata(pdev);
 
+	pinctrl_pm_select_default_state(&pdev->dev);
+
 	ret = fsl_qspi_clk_prep_enable(q);
 	if (ret)
 		return ret;
-- 
1.8.0

